test/patterns/null_check/test_null_check.py

```diff
@@ -20,23 +20,68 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-import os
-from pathlib import Path
 from unittest import TestCase
 
+from javalang.parser import Parser
+from javalang.tokenizer import tokenize
+
 from aibolit.patterns.null_check.null_check import NullCheck
 
 
 class TestNullCheck(TestCase):
-    cur_file_dir = Path(os.path.realpath(__file__)).parent
-    testClass = NullCheck()
+    def setUp(self):
+        self.pattern = NullCheck()
 
     def test_null_check(self):
-        file = Path(self.cur_file_dir, 'NullCheck.java')
+        snippet = """\
+        if (this.z == null) { // here!
+            throw new RuntimeException("oops");
+        }
+        """
 
-        self.assertEqual(self.testClass.value(file), [5])
+        self.assertEqual(
+            self.pattern._traverse_node(_parser(snippet).parse_block_statement()), [1]
```

Comment:
@tribals I think it is better to:
`NullCheck()._traverse_node(_parser(snippet).parse_block_statement()), [1]` as each test method is a dedicated test unit.
It is also not good to have this  `_traverse_node` method calling here.

PS: Actually each pattern should accept the `snippet `or `filename `or etc via constructor, but not the `value `method, but this is out of scope of this PR.

Reply:
There is no need to create `NullCheck` explicitly in each test method since `setUp` is called once per every test method. Please, see: https://docs.python.org/3/library/unittest.html#organizing-tests.

I agree, though, that it is not clear from source code itself.

> It is also not good to have this `_traverse_node` method calling here.

I absolutely agree. A method which name is prefixed with underscore considered a private one, and is not allowed to call directly. But, since there is only one possible way to pass already parsed AST to pattern now, I suggest to leave it as is until we will fix all patterns.

Reply:
> There is no need to create `NullCheck` explicitly in each test method since `setUp` is called once per every test method. Thus, `self.pattern` is a new one in each test. Please, see: https://docs.python.org/3/library/unittest.html#organizing-tests.
> 
> I agree, though, that it is not clear from source code itself.
> 
> > It is also not good to have this `_traverse_node` method calling here.
> 
> I absolutely agree. A method which name is prefixed with underscore considered a private one, and is not allowed to call directly. But, since there is only one possible way to pass already parsed AST to pattern now, I suggest to leave it as is until we will fix all patterns.

@tribals I know that `setup` does it for each test method, but I personally(not only me) find that setup method is very confusing and brings more confusions. 
I think it is better to create pattern object in each test method  separately.
@acheshkov WDYT ?

Reply:
Well, personally I find the same about setup methods, too. So, it is simple to me to follow you advice.